### PR TITLE
Make validator::new return ValidatorError.

### PR DESCRIPTION
**Pull request by wen-coding**
In the future, we need to match specific error type when wen_restart returns.

Make Validator::new() return ValidatorError instead of a String Error.

### CODE REVIEW

File: core/src/validator.rs [hunk 1]
```diff
@@ -2797,8 +2816,10 @@ mod tests {
                 &cluster_info,
                 rpc_override_health_check,
                 &start_progress,
-            ),
-            Err(ValidatorError::BadExpectedBankHash)
+            )
+            .unwrap_err()
+            .to_string(),
+            "Bad expected bank hash"
```
Comment (carllin) [inline core/src/validator.rs]: Can just matches!() on the validator error `Err(ValidatorError::BadExpectedBankHash)`

**Comment by carllin** [comment]
I think we need to parse the validator error and return a new code here https://github.com/anza-xyz/agave/blob/61473bc5d9b5f0148ee71fb4aea669997343f643/validator/src/main.rs#L2002-L2003 so operators can switch on the return value

File: core/src/validator.rs [hunk 2]
```diff
@@ -2765,8 +2782,10 @@ mod tests {
                 &cluster_info,
                 rpc_override_health_check.clone(),
                 &start_progress,
-            ),
-            Err(ValidatorError::NotEnoughLedgerData)
+            )
+            .unwrap_err()
+            .to_string(),
+            "Ledger does not have enough data to wait for supermajority"
```
Comment (carllin) [inline core/src/validator.rs]: Can just matches!() on the validator error

**Comment by wen-coding** [comment]
> I think we need to parse the validator error and return a new code here
> 
> https://github.com/anza-xyz/agave/blob/61473bc5d9b5f0148ee71fb4aea669997343f643/validator/src/main.rs#L2002-L2003
> 
> so operators can switch on the return value

Planned to that in a separate PR to make this PR small.

File: core/src/validator.rs (hunk 1 above)
Comment (wen-coding) [reply core/src/validator.rs]: Done.

File: core/src/validator.rs (hunk 2 above)
Comment (wen-coding) [reply core/src/validator.rs]: Done.

**Review by carllin — APPROVED** [review]

**Pull request merged**